tests/history.spec.ts

```diff
@@ -0,0 +1,136 @@
+import { expect, test } from '@playwright/test';
+import { Affiliations, History, MainText, References, SingleTextField } from './page-objects';
+
+test.describe('history', () => {
+  let history: History;
+
+  test.beforeEach(async ({ page }) => {
+    await page.goto('http://localhost:3000/?articleId=54296');
+    history = new History(page);
+    await page.waitForSelector('#title');
+  });
+
+  test.describe('undo', () => {
+    test('undo title changes', async ({ page }) => {
+      const field = new SingleTextField(page, 'title');
+      const content = 'Infinity welcomes careful drivers';
+      await field.setField('');
+      await field.typeField(content);
+      await field.assertField(content);
+      await history.undoNTimes(10);
+      await field.assertField('Infinity welcomes caref');
+    });
+
+    test('undo adding affiliation', async ({ page }) => {
+      const affiliations = new Affiliations(page);
+      await affiliations.addAffiliation({ institution: 'JMC', city: 'Ganymede', country: 'Jupiter' });
+      const initialCount = await affiliations.affiliationCount();
+      await history.undoNTimes(2);
```

Comment:
Is this the case that you were talking about where you need to undo twice to actually step back once in the history? If so, might be worth adding a `FIXME` comment just to note the workaround for future reference.